### ISSUE TITLE
Coupon messages don't show in the woocommerce-notices-wrapper section

### ISSUE THREAD

**Issue by dealdot** [woothemes/woocommerce#53534]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The coupon message (such as does'nt exist or Please enter a coupon code.) does't show in the woocommerce-notices-wrapper section since woocommerce version 9.4, it works in 9.3.3 and belows. I  using classic cart and classic checkout for theme developing.

### Expected behavior

![Image](https://github.com/user-attachments/assets/b0d19879-f38f-42ad-84e4-f9ab3e8a153a)

### Actual behavior

![Image](https://github.com/user-attachments/assets/5be57fcd-c291-415f-9d2b-90d6c7027ba4)

### Steps to reproduce

1. using wooommerce 9.4
2. using classic cart and classic checkout

### WordPress Environment

wordpress: 6.7.1
woocommerce: 9.4.2

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by senadir** [comment]
Hey! In https://github.com/woocommerce/woocommerce/pull/48738 coupon error moved from being a notice to being inline in the coupon input

![Image](https://github.com/user-attachments/assets/af1700dc-c959-46a6-aead-51df3c644aa1)


Are you running into problems seeing it on the input? Which theme you're testing with as well?

**Comment by mahfuzurwp** [comment]
Hi, I wanted to provide feedback on the issue. The coupon error message in the Storefront theme is now displayed as grey text, significantly impacting the user experience. It’s not very noticeable and can easily be overlooked by users. Moving the error message inline in the coupon input is not a great UX choice, as it makes the message less prominent.

I’ve attached a screenshot for reference. I hope this feedback helps!

![Image](https://github.com/user-attachments/assets/94b3caf5-c8af-4f32-adaa-80a45a205840)

**Comment by senadir** [comment]
Moving error inline lines up to our UX understanding that an error should be closer to the field that caused it, a person clicking apply coupon i actively focusing on the field and interacting with it, showing a notice top page is distracting. This is also more accessible for screen reader.

I do agree about the color issue @Manussakis @dkotter can we look into that?

**Comment by Manussakis** [comment]
@senadir This issue with the coupon error message on Storefront [was flagged](https://github.com/woocommerce/woocommerce/pull/48738#pullrequestreview-2204533147) by @nielslange on the https://github.com/woocommerce/woocommerce/pull/48738 PR thread.

Then I replied with [this comment](https://github.com/woocommerce/woocommerce/pull/48738#issuecomment-2255747278) and finally [this was the approach](https://github.com/woocommerce/woocommerce/pull/48738#issuecomment-2273759321) that @mikejolley suggested following.

### TL;DR

The Storefront theme doesn't load the Woo core stylesheet, which contains the CSS rules that make the error message red and well-positioned. Therefore, the fix should be addressed on the Storefront side, if I am not mistaken.

I'd like to ask what the steps are to communicate with the Storefront team to relay this kind of information before getting regressions next time we face a similar situation.

**Comment by senadir** [comment]
we are the storefront team, you can go ahead and make a PR there though we should look into reusing some existing class that works with different themea

**Comment by Manussakis** [comment]
Thanks, @senadir! 
Here is the PR: https://github.com/woocommerce/storefront/pull/2178

**Comment by dealdot** [comment]
Hello, @senadir thanks for your responding about my issue, but when the coupon was applied successfully or removed , it remains the old version(<9.3) top notice,  As follows: 

**applied successfully**

![Image](https://github.com/user-attachments/assets/d5475195-b3f6-4603-946e-db9592d08078)

**removed**

![Image](https://github.com/user-attachments/assets/c06c7c95-cfe1-45f6-af0a-fb8258e11a47)


**So, in my opinion the message prompts' UI is inconsistent.  While, as the checkout page,  it is reasonable to display coupon validation messages inline.**

![Image](https://github.com/user-attachments/assets/443a860b-9d05-437b-961c-be2c8e2886be)

Thanks.

**Comment by dealdot** [comment]
**And if any one who are using classic cart may find there is no validation message, then pay attention to the code as follows, the class name of "coupon" must be there.**

![Image](https://github.com/user-attachments/assets/005f68cb-a566-4625-bf06-67e453b639e0)

**Comment by Manussakis** [comment]
Thanks @dealdot for sharing your thoughts!

> So, in my opinion the message prompts' UI is inconsistent.

The inline error message is for situations when the user tries to add a coupon code but the action fails for some reason, for example:
- The coupon code has been misspelled.
- The coupon has been already applied.

In these cases, keeping the error message close to the input that causes the error is better for UX, especially because the input gets focused on waiting for the next user interaction.

On the other hand, when users successfully apply or remove a coupon, there is no additional action needed, so sticking to the notice at the top makes sense.

The inline error message serves a different purpose than the success messages.

**Comment by imanish003** [comment]
Update: PR #2178 has been merged. A new version of Storefront containing this fix will be released soon.

**Comment by imanish003** [comment]
Update: PR [#2178](https://github.com/woocommerce/storefront/pull/2178) has been merged. A new version of Storefront containing this fix will be released soon. We can always reopen this issue if we need to.

**Comment by sarun007** [comment]
Hi @imanish003 

I can see that you mentioned this issue has been merged with PR https://github.com/woocommerce/woocommerce/issues/2178. However, I can't find anything mentioned about the coupon error message in that issue. Additionally, I noticed that https://github.com/woocommerce/woocommerce/issues/2178 is already closed.

Are you currently working on the coupon error issue?

**Comment by nielslange** [comment]
> I can see that you mentioned this issue has been merged with PR https://github.com/woocommerce/woocommerce/issues/2178. However, I can't find anything mentioned about the coupon error message in that issue. Additionally, I noticed that https://github.com/woocommerce/woocommerce/issues/2178 is already closed.

@sarun007 The issue https://github.com/woocommerce/storefront/pull/2178 is in the Storefront repo. I corrected the link in @imanish003's [comment](https://github.com/woocommerce/woocommerce/issues/53534#issuecomment-2579957134) above. Sorry for the confusion.
